src/pudl/transform/nrelatb.py

```diff
@@ -0,0 +1,368 @@
+"""Transform NREL ATB data into well normalized, cleaned tables."""
+
+from typing import Self
+
+import pandas as pd
+from dagster import asset
+from pydantic import BaseModel, field_validator
+
+import pudl.helpers as helpers
+
+IDX_ALL = [
+    "report_year",
+    "core_metric_case",
+    "core_metric_variable_year",
+    "cost_recovery_period_years",
+    "scenario_atb",
+    "technology_description",
+    "technology_description_detail_1",
+    "technology_description_detail_2",
+    "core_metric_parameter",
+]
+"""Expected primary key columns for the raw nrelatb data.
+
+The normalized core tables we are trying to build will result in a subset of these columns.
+"""
+
+
+class TableNormalizer(BaseModel):
+    """Info needed to convert a selection of a the raw NREL table into a normalized table."""
```

Comment:
Now that thousands of GSoD applicants are looking at our every line of code, I'm gonna be nitpicky about typos 😉 

```suggestion
    """Info needed to convert a selection of the raw NREL table into a normalized table."""
```

Reply:
Separately from the typo, it's probably useful to describe somewhere the funky dissection you're doing where certain data columns correspond with certain index columns and certain `metric`s...

Reply:
okay i think i've added this context into the appropriate class/function docs.